subsys/bluetooth/mesh/rpl.c

```diff
@@ -344,11 +350,7 @@ void bt_mesh_rpl_pending_store(uint16_t addr)
 			continue;
 		}
 
-		if (atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
-			store_pending_rpl(&replay_list[i]);
-		} else {
-			clear_rpl(&replay_list[i]);
-		}
+		store_pending_rpl(&replay_list[i]);
```

Comment:
I dont think so， bt mesh reset will call bt mesh rpl clear

Reply:
All of this process in sysworkq, same as previously. Not change task schedule behavior.

Reply:
It will call `void bt_mesh_settings_store_pending(void)` that causes `store_pending_rpl(&replay_list[i]);` since provisioning state is not checked anymore. 
In the best way settings will try to clear some area twice.
In the worst case something can sneak after transport layer reset and before settings reset into RPL RAM representation.

Reply:
static void store_rpl(struct bt_mesh_rpl *entry)
{
	struct rpl_val rpl;
	char path[18];
	int err;

	if (!entry->src) {
		return;
	}

Reply:
If rpl clear pre proceed, src will be zero, so there are not store.

When rpl store pre proceed, rpl clear will deleted.

So there are no problem.

Reply:
> It will call `void bt_mesh_settings_store_pending(void)` that causes `store_pending_rpl(&replay_list[i]);` since provisioning state is not checked anymore.

`bt_mesh_reset --> bt_mesh_trans_reset-->bt_mesh_rpl_clear
`
Also add check for store rpl:

``` C

	if (!atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
		return;
	}
````

Reply:
This is kind of workaround that prevents coming into cleaning procedure twice but you correctly found that `store_rpl` will reject cleaning the empty entries. 
However, we get situation when during reset the stack tries to clean up RPL twice.
1. via transport layer
2. via settings
Is it ok?

Reply:
Previously in transport layer stack scheduled task and handled it in settings. Kind of collaboration between them :)